include/amber/amber.h

```diff
@@ -116,6 +117,9 @@ struct Options {
   bool disable_spirv_validation;
   /// Delegate implementation
   Delegate* delegate;
+
+  std::function<amber::Result(std::string, amber::BufferInfo*)>
```

Comment:
Move the callback method up to the Delegate class above instead of as an option. (Please add a comment as well)